hashintel/hash
# Various UX improvements for entity editor

## 🌟 What is the purpose of this PR?

This PR fixes few bugs and improves UX around entity editor. List of the changes are listed below.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543026679373/1203842233513186/f) _(internal)_

## 🔍 What does this change?

- prevent json editor default value being `null`
- skip edit mode for no-editor data types like null
- improve selection of no-editor types like null
- bump `@glideapps/glide-data-grid` to v5.2.1
- improve width & resize of the table columns
- prevent rendering empty arrays as empty-list chip
- use "add value" if the first value is being added
- fix invisible grid overlay inside drawer by updating `z-index` of the overlay

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Try the changes listed above
2. Confirm that changes works as expected

## 📹 Demo

https://user-images.githubusercontent.com/39553853/215032351-2f0b9210-1543-4cd1-9f2f-85b06003aac2.mp4
